? wxWindows/wxgtkd-2.3-config
? wxWindows/wxgtk-2.3-config
? wxWindows/lexer.c
? wxWindows/parser.c
? wxWindows/contrib/config.log
? wxWindows/contrib/config.status
? wxWindows/contrib/samples/Makefile
? wxWindows/contrib/samples/applet/Makefile
? wxWindows/contrib/samples/canvas/Makefile
? wxWindows/contrib/samples/canvas/simple/Makefile
? wxWindows/contrib/samples/canvas/test/Makefile
? wxWindows/contrib/samples/fl/Makefile
? wxWindows/contrib/samples/fl/fl_demo1/Makefile
? wxWindows/contrib/samples/fl/fl_demo2/Makefile
? wxWindows/contrib/samples/fl/fl_sample1/Makefile
? wxWindows/contrib/samples/fl/fl_sample2/Makefile
? wxWindows/contrib/samples/fl/fl_sample3/Makefile
? wxWindows/contrib/samples/gizmos/Makefile
? wxWindows/contrib/samples/gizmos/dynsash/Makefile
? wxWindows/contrib/samples/gizmos/dynsash_switch/Makefile
? wxWindows/contrib/samples/gizmos/editlbox/Makefile
? wxWindows/contrib/samples/gizmos/multicell/Makefile
? wxWindows/contrib/samples/gizmos/splittree/Makefile
? wxWindows/contrib/samples/mmedia/Makefile
? wxWindows/contrib/samples/ogl/Makefile
? wxWindows/contrib/samples/ogl/ogledit/Makefile
? wxWindows/contrib/samples/ogl/studio/Makefile
? wxWindows/contrib/samples/plot/Makefile
? wxWindows/contrib/samples/stc/Makefile
? wxWindows/contrib/samples/xrc/Makefile
? wxWindows/contrib/src/Makefile
? wxWindows/contrib/src/applet/Makefile
? wxWindows/contrib/src/canvas/Makefile
? wxWindows/contrib/src/fl/Makefile
? wxWindows/contrib/src/gizmos/Makefile
? wxWindows/contrib/src/mmedia/Makefile
? wxWindows/contrib/src/net/Makefile
? wxWindows/contrib/src/ogl/Makefile
? wxWindows/contrib/src/plot/Makefile
? wxWindows/contrib/src/stc/Makefile
? wxWindows/contrib/src/xrc/Makefile
? wxWindows/contrib/utils/Makefile
? wxWindows/contrib/utils/convertrc/Makefile
? wxWindows/contrib/utils/wxrc/Makefile
? wxWindows/contrib/utils/wxrcedit/Makefile
? wxWindows/demos/config.log
? wxWindows/demos/config.status
? wxWindows/demos/Makefile
? wxWindows/demos/bombs/Makefile
? wxWindows/demos/dbbrowse/Makefile
? wxWindows/demos/forty/Makefile
? wxWindows/demos/fractal/Makefile
? wxWindows/demos/life/Makefile
? wxWindows/demos/poem/Makefile
? wxWindows/lib/wx
? wxWindows/lib/libwx_gtkd-2.3.so.2.0.0
? wxWindows/lib/libwx_gtkd_gl-2.3.so.2.0.0
? wxWindows/samples/config.log
? wxWindows/samples/config.status
? wxWindows/samples/Makefile
? wxWindows/samples/artprov/Makefile
? wxWindows/samples/calendar/Makefile
? wxWindows/samples/caret/Makefile
? wxWindows/samples/checklst/Makefile
? wxWindows/samples/config/Makefile
? wxWindows/samples/console/Makefile
? wxWindows/samples/controls/Makefile
? wxWindows/samples/db/Makefile
? wxWindows/samples/dialogs/Makefile
? wxWindows/samples/dialup/Makefile
? wxWindows/samples/dnd/Makefile
? wxWindows/samples/docview/Makefile
? wxWindows/samples/docvwmdi/Makefile
? wxWindows/samples/dragimag/Makefile
? wxWindows/samples/drawing/Makefile
? wxWindows/samples/dynamic/Makefile
? wxWindows/samples/erase/Makefile
? wxWindows/samples/event/Makefile
? wxWindows/samples/exec/Makefile
? wxWindows/samples/font/Makefile
? wxWindows/samples/grid/Makefile
? wxWindows/samples/help/Makefile
? wxWindows/samples/html/Makefile
? wxWindows/samples/html/about/Makefile
? wxWindows/samples/html/help/Makefile
? wxWindows/samples/html/helpview/Makefile
? wxWindows/samples/html/printing/Makefile
? wxWindows/samples/html/test/Makefile
? wxWindows/samples/html/virtual/Makefile
? wxWindows/samples/html/widget/Makefile
? wxWindows/samples/html/zip/Makefile
? wxWindows/samples/image/Makefile
? wxWindows/samples/internat/Makefile
? wxWindows/samples/ipc/Makefile
? wxWindows/samples/joytest/Makefile
? wxWindows/samples/keyboard/Makefile
? wxWindows/samples/layout/Makefile
? wxWindows/samples/listctrl/Makefile
? wxWindows/samples/mdi/Makefile
? wxWindows/samples/memcheck/Makefile
? wxWindows/samples/menu/Makefile
? wxWindows/samples/minifram/Makefile
? wxWindows/samples/minimal/Makefile
? wxWindows/samples/mobile/Makefile
? wxWindows/samples/mobile/wxedit/Makefile
? wxWindows/samples/newgrid/Makefile
? wxWindows/samples/newgrid/griddemo.d
? wxWindows/samples/newgrid/griddemo
? wxWindows/samples/notebook/Makefile
? wxWindows/samples/opengl/Makefile
? wxWindows/samples/opengl/cube/Makefile
? wxWindows/samples/opengl/isosurf/Makefile
? wxWindows/samples/opengl/penguin/Makefile
? wxWindows/samples/png/Makefile
? wxWindows/samples/printing/Makefile
? wxWindows/samples/proplist/Makefile
? wxWindows/samples/propsize/Makefile
? wxWindows/samples/resource/Makefile
? wxWindows/samples/richedit/Makefile
? wxWindows/samples/rotate/Makefile
? wxWindows/samples/sashtest/Makefile
? wxWindows/samples/scroll/Makefile
? wxWindows/samples/scrollsub/Makefile
? wxWindows/samples/sockets/Makefile
? wxWindows/samples/splitter/Makefile
? wxWindows/samples/statbar/Makefile
? wxWindows/samples/text/Makefile
? wxWindows/samples/thread/Makefile
? wxWindows/samples/toolbar/Makefile
? wxWindows/samples/treectrl/Makefile
? wxWindows/samples/treelay/Makefile
? wxWindows/samples/typetest/Makefile
? wxWindows/samples/validate/Makefile
? wxWindows/samples/widgets/Makefile
? wxWindows/samples/wizard/Makefile
? wxWindows/utils/config.log
? wxWindows/utils/config.status
? wxWindows/utils/Makefile
? wxWindows/utils/HelpGen/Makefile
? wxWindows/utils/HelpGen/src/Makefile
? wxWindows/utils/dialoged/Makefile
? wxWindows/utils/emulator/Makefile
? wxWindows/utils/emulator/src/Makefile
? wxWindows/utils/hhp2cached/Makefile
? wxWindows/utils/makegen/Makefile
? wxWindows/utils/tex2rtf/Makefile
? wxWindows/utils/tex2rtf/src/Makefile
Index: wxWindows/src/generic/grid.cpp
===================================================================
RCS file: /home/wxcvs/wxWindows/src/generic/grid.cpp,v
retrieving revision 1.197
diff -r1.197 grid.cpp
3805c3805,3808
<     return m_colWidths.IsEmpty() ? m_defaultColWidth : m_colWidths[col];
---
>     if (col == -1)
> 	return m_rowLabelWidth;
>     else
> 	return m_colWidths.IsEmpty() ? m_defaultColWidth : m_colWidths[col];
3810,3811c3813,3817
<     return m_colRights.IsEmpty() ? col * m_defaultColWidth
<                                  : m_colRights[col] - m_colWidths[col];
---
>     if (col == -1)
> 	return 0;
>     else
> 	return m_colRights.IsEmpty() ? col * m_defaultColWidth
>                                      : m_colRights[col] - m_colWidths[col];
3816,3817c3822,3827
<     return m_colRights.IsEmpty() ? (col + 1) * m_defaultColWidth
<                                  : m_colRights[col];
---
>     if (col == -1)
> 	return m_colRights.IsEmpty() ? m_defaultColWidth
>                                      : m_colRights[0] - m_colWidths[0];
>     else
> 	return m_colRights.IsEmpty() ? (col + 1) * m_defaultColWidth
>                                      : m_colRights[col];
3822c3832,3835
<     return m_rowHeights.IsEmpty() ? m_defaultRowHeight : m_rowHeights[row];
---
>     if (row == -1)
> 	return m_colLabelHeight;
>     else
> 	return m_rowHeights.IsEmpty() ? m_defaultRowHeight : m_rowHeights[row];
3827,3828c3840,3844
<     return m_rowBottoms.IsEmpty() ? row * m_defaultRowHeight
<                                   : m_rowBottoms[row] - m_rowHeights[row];
---
>     if (row == -1)
> 	return 0;
>     else
> 	return m_rowBottoms.IsEmpty() ? row * m_defaultRowHeight
> 	                              : m_rowBottoms[row] - m_rowHeights[row];
3833,3834c3849,3854
<     return m_rowBottoms.IsEmpty() ? (row + 1) * m_defaultRowHeight
<                                   : m_rowBottoms[row];
---
>     if (row == -1)
> 	return m_rowBottoms.IsEmpty() ? m_defaultRowHeight
> 	                              : m_rowBottoms[0] - m_rowHeights[0];
>     else
> 	return m_rowBottoms.IsEmpty() ? (row + 1) * m_defaultRowHeight
>                                       : m_rowBottoms[row];
6727,6728c6747,6753
<             rect.x--;
<             rect.y--;
---
> 	    // However, don't allow x or y to become negative
> 	    // since the SetSize() method interprets that as 
> 	    // "don't change."
> 	    if (rect.x > 0)
>   	        rect.x--;
> 	    if (rect.y > 0)
> 	        rect.y--;
6859a6885,6887
>     if ( coord < BorderArray[i_min])
>         return 0;
> 

 	  	 
